Auto vacuum process can cause transaction timeout

### Description

On performance environment testing HCS at 10K TPS, we saw multiple latency spikes. Investigation revealed that the importer transaction timed out due to abnormally high insert times for the `transaction` table.

### Steps to reproduce

HCS 10K TPS

### Additional context

## 2023-08-03

PostgreSQL
```bash
2023-08-03 12:04:01.126 UTC [1257452]: [2-1] db=,user= LOG:  automatic vacuum of table "mirror_node.public.transaction": index scans: 0
pages: 0 removed, 7053942 remain, 0 skipped due to pins, 3486793 skipped frozen
tuples: 0 removed, 150983807 remain, 0 are dead but not yet removable, oldest xmin: 31844956
index scan not needed: 0 pages from table (0.00% of total) had 0 dead item identifiers removed
avg read rate: 21.915 MB/s, avg write rate: 12.567 MB/s
buffer usage: 3723226 hits, 3413747 misses, 1957558 dirtied
WAL usage: 5651962 records, 1957562 full page images, 16438808506 bytes
system usage: CPU: user: 44.44 s, system: 122.82 s, elapsed: 1216.95 s
```

Importer
```bash
2023-08-03T12:04:01.450679437Z stdout F 2023-08-03T06:04:01.450-0600 INFO scheduling-4 c.h.m.i.p.b.BatchInserter Copied 80006 rows to crypto_transfer table in 339.2 ms 
2023-08-03T12:04:01.69353372Z stdout F 2023-08-03T06:04:01.693-0600 INFO scheduling-4 c.h.m.i.p.b.BatchInserter Copied 19990 rows to topic_message table in 242.6 ms 
2023-08-03T12:04:30.724645084Z stdout F 2023-08-03T06:04:30.724-0600 INFO scheduling-4 c.h.m.i.p.b.BatchInserter Copied 20000 rows to transaction table in 29.03 s 
2023-08-03T12:04:30.73414135Z stdout F 2023-08-03T06:04:30.733-0600 INFO scheduling-4 c.h.m.i.p.b.BatchInserter Copied 34 rows to entity_temp table in 9.327 ms 
2023-08-03T12:04:30.734216147Z stdout F 2023-08-03T06:04:30.733-0600 INFO scheduling-4 c.h.m.i.p.r.e.s.SqlEntityListener Completed batch inserts in 29.62 s 
2023-08-03T12:04:30.770757919Z stdout F 2023-08-03T06:04:30.770-0600 INFO scheduling-4 c.h.m.i.p.b.BatchInserter Copied 6074 rows to crypto_transfer table in 29.39 ms 
2023-08-03T12:04:30.790606125Z stdout F 2023-08-03T06:04:30.790-0600 INFO scheduling-4 c.h.m.i.p.b.BatchInserter Copied 1516 rows to topic_message table in 19.68 ms 
2023-08-03T12:04:33.060059342Z stdout F 2023-08-03T06:04:33.059-0600 INFO scheduling-4 c.h.m.i.p.b.BatchInserter Copied 1518 rows to transaction table in 2.269 s 
2023-08-03T12:04:33.067990591Z stdout F 2023-08-03T06:04:33.067-0600 INFO scheduling-4 c.h.m.i.p.b.BatchInserter Copied 28 rows to entity_temp table in 7.794 ms 
2023-08-03T12:04:33.068233435Z stdout F 2023-08-03T06:04:33.068-0600 INFO scheduling-4 c.h.m.i.p.r.e.s.SqlEntityListener Completed batch inserts in 2.327 s 
2023-08-03T12:04:33.128371065Z stdout F 2023-08-03T06:04:33.122-0600 ERROR scheduling-4 c.h.m.i.p.r.RecordFileParser Error parsing file 2023-08-03T12_03_54.004371118Z.rcd.gz after 32.11 s org.springframework.orm.jpa.JpaSystemException: transaction timeout expired
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:320)
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:229)
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.translateExceptionIfPossible(AbstractEntityManagerFactoryBean.java:550)
	at org.springframework.dao.support.ChainedPersistenceExceptionTranslator.translateExceptionIfPossible(ChainedPersistenceExceptionTranslator.java:61)
	at org.springframework.dao.support.DataAccessUtils.translateIfNecessary(DataAccessUtils.java:242)
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:152)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.invoke(CrudMethodMetadataPostProcessor.java:164)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:244)
	at jdk.proxy2/jdk.proxy2.$Proxy255.save(Unknown Source)
	at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.onEnd(SqlEntityListener.java:230)
	at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.onEnd(SqlEntityListener.java:90)
	at com.hedera.mirror.importer.parser.record.entity.CompositeRecordStreamFileListener.onEnd(CompositeRecordStreamFileListener.java:44)
	at com.hedera.mirror.importer.parser.record.entity.CompositeRecordStreamFileListener.onEnd(CompositeRecordStreamFileListener.java:29)
	at com.hedera.mirror.importer.parser.record.RecordFileParser.doParse(RecordFileParser.java:145)
	at com.hedera.mirror.importer.parser.record.RecordFileParser.doParse(RecordFileParser.java:47)
	at com.hedera.mirror.importer.parser.AbstractStreamFileParser.parse(AbstractStreamFileParser.java:79)
	at com.hedera.mirror.importer.parser.record.RecordFileParser.parse(RecordFileParser.java:118)
	at jdk.internal.reflect.GeneratedMethodAccessor140.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:343)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:756)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:391)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:756)
	at org.springframework.retry.interceptor.RetryOperationsInterceptor$1.doWithRetry(RetryOperationsInterceptor.java:102)
	at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:329)
	at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:209)
	at org.springframework.retry.interceptor.RetryOperationsInterceptor.invoke(RetryOperationsInterceptor.java:135)
	at org.springframework.retry.annotation.AnnotationAwareRetryOperationsInterceptor.invoke(AnnotationAwareRetryOperationsInterceptor.java:160)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:756)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708)
	at com.hedera.mirror.importer.parser.record.RecordFileParser$$SpringCGLIB$$0.parse(<generated>)
	at com.hedera.mirror.importer.config.MessagingConfiguration.lambda$integrationFlow$1(MessagingConfiguration.java:107)
	at org.springframework.integration.handler.LambdaMessageProcessor.invokeMethod(LambdaMessageProcessor.java:208)
	at org.springframework.integration.handler.LambdaMessageProcessor.processMessage(LambdaMessageProcessor.java:121)
	at org.springframework.integration.handler.ServiceActivatingHandler.handleRequestMessage(ServiceActivatingHandler.java:95)
	at org.springframework.integration.handler.AbstractReplyProducingMessageHandler.handleMessageInternal(AbstractReplyProducingMessageHandler.java:136)
	at org.springframework.integration.handler.AbstractMessageHandler.doHandleMessage(AbstractMessageHandler.java:105)
	at org.springframework.integration.handler.AbstractMessageHandler.handleWithMetrics(AbstractMessageHandler.java:90)
	at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:70)
	at org.springframework.integration.endpoint.PollingConsumer.handleMessage(PollingConsumer.java:158)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.messageReceived(AbstractPollingEndpoint.java:474)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.doPoll(AbstractPollingEndpoint.java:460)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.pollForMessage(AbstractPollingEndpoint.java:412)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$createPoller$4(AbstractPollingEndpoint.java:348)
	at org.springframework.integration.util.ErrorHandlingTaskExecutor.lambda$execute$0(ErrorHandlingTaskExecutor.java:57)
	at org.springframework.core.task.SyncTaskExecutor.execute(SyncTaskExecutor.java:50)
	at org.springframework.integration.util.ErrorHandlingTaskExecutor.execute(ErrorHandlingTaskExecutor.java:55)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$createPoller$5(AbstractPollingEndpoint.java:341)
	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)
	at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:96)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: org.hibernate.TransactionException: transaction timeout expired
	at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.determineRemainingTransactionTimeOutPeriod(JdbcCoordinatorImpl.java:262)
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl$StatementPreparationTemplate.setStatementTimeout(StatementPreparerImpl.java:199)
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl$StatementPreparationTemplate.prepareStatement(StatementPreparerImpl.java:178)
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl.prepareStatement(StatementPreparerImpl.java:76)
	at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.lambda$list$0(JdbcSelectExecutorStandardImpl.java:102)
	at org.hibernate.sql.results.jdbc.internal.DeferredResultSetAccess.executeQuery(DeferredResultSetAccess.java:226)
	at org.hibernate.sql.results.jdbc.internal.DeferredResultSetAccess.getResultSet(DeferredResultSetAccess.java:163)
	at org.hibernate.sql.results.jdbc.internal.JdbcValuesResultSetImpl.advanceNext(JdbcValuesResultSetImpl.java:254)
	at org.hibernate.sql.results.jdbc.internal.JdbcValuesResultSetImpl.processNext(JdbcValuesResultSetImpl.java:134)
	at org.hibernate.sql.results.jdbc.internal.AbstractJdbcValues.next(AbstractJdbcValues.java:19)
	at org.hibernate.sql.results.internal.RowProcessingStateStandardImpl.next(RowProcessingStateStandardImpl.java:66)
	at org.hibernate.sql.results.spi.ListResultsConsumer.consume(ListResultsConsumer.java:184)
	at org.hibernate.sql.results.spi.ListResultsConsumer.consume(ListResultsConsumer.java:33)
	at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.doExecuteQuery(JdbcSelectExecutorStandardImpl.java:361)
	at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.executeQuery(JdbcSelectExecutorStandardImpl.java:168)
	at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.list(JdbcSelectExecutorStandardImpl.java:93)
	at org.hibernate.sql.exec.spi.JdbcSelectExecutor.list(JdbcSelectExecutor.java:31)
	at org.hibernate.loader.ast.internal.SingleIdLoadPlan.load(SingleIdLoadPlan.java:146)
	at org.hibernate.loader.ast.internal.SingleIdLoadPlan.load(SingleIdLoadPlan.java:118)
	at org.hibernate.loader.ast.internal.SingleIdEntityLoaderStandardImpl.load(SingleIdEntityLoaderStandardImpl.java:73)
	at org.hibernate.persister.entity.AbstractEntityPersister.doLoad(AbstractEntityPersister.java:3481)
	at org.hibernate.persister.entity.AbstractEntityPersister.load(AbstractEntityPersister.java:3471)
	at org.hibernate.event.internal.DefaultLoadEventListener.loadFromDatasource(DefaultLoadEventListener.java:581)
	at org.hibernate.event.internal.DefaultLoadEventListener.loadFromCacheOrDatasource(DefaultLoadEventListener.java:567)
	at org.hibernate.event.internal.DefaultLoadEventListener.load(DefaultLoadEventListener.java:536)
	at org.hibernate.event.internal.DefaultLoadEventListener.doLoad(DefaultLoadEventListener.java:529)
	at org.hibernate.event.internal.DefaultLoadEventListener.load(DefaultLoadEventListener.java:202)
	at org.hibernate.event.internal.DefaultLoadEventListener.loadWithRegularProxy(DefaultLoadEventListener.java:282)
	at org.hibernate.event.internal.DefaultLoadEventListener.proxyOrLoad(DefaultLoadEventListener.java:237)
	at org.hibernate.event.internal.DefaultLoadEventListener.doOnLoad(DefaultLoadEventListener.java:106)
	at org.hibernate.event.internal.DefaultLoadEventListener.onLoad(DefaultLoadEventListener.java:78)
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:138)
	at org.hibernate.internal.SessionImpl.fireLoadNoChecks(SessionImpl.java:1231)
	at org.hibernate.internal.SessionImpl.fireLoad(SessionImpl.java:1219)
	at org.hibernate.loader.internal.IdentifierLoadAccessImpl.doLoad(IdentifierLoadAccessImpl.java:194)
	at org.hibernate.loader.internal.IdentifierLoadAccessImpl.lambda$load$1(IdentifierLoadAccessImpl.java:160)
	at org.hibernate.loader.internal.IdentifierLoadAccessImpl.perform(IdentifierLoadAccessImpl.java:107)
	at org.hibernate.loader.internal.IdentifierLoadAccessImpl.load(IdentifierLoadAccessImpl.java:160)
	at org.hibernate.internal.SessionImpl.get(SessionImpl.java:1024)
	at org.hibernate.event.internal.DefaultMergeEventListener.lambda$entityIsDetached$0(DefaultMergeEventListener.java:334)
	at org.hibernate.engine.spi.LoadQueryInfluencers.fromInternalFetchProfile(LoadQueryInfluencers.java:79)
	at org.hibernate.event.internal.DefaultMergeEventListener.entityIsDetached(DefaultMergeEventListener.java:332)
	at org.hibernate.event.internal.DefaultMergeEventListener.merge(DefaultMergeEventListener.java:149)
	at org.hibernate.event.internal.DefaultMergeEventListener.doMerge(DefaultMergeEventListener.java:142)
	at org.hibernate.event.internal.DefaultMergeEventListener.onMerge(DefaultMergeEventListener.java:126)
	at org.hibernate.event.internal.DefaultMergeEventListener.onMerge(DefaultMergeEventListener.java:80)
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:127)
	at org.hibernate.internal.SessionImpl.fireMerge(SessionImpl.java:848)
	at org.hibernate.internal.SessionImpl.merge(SessionImpl.java:834)
	at jdk.internal.reflect.GeneratedMethodAccessor139.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.springframework.orm.jpa.SharedEntityManagerCreator$SharedEntityManagerInvocationHandler.invoke(SharedEntityManagerCreator.java:311)
	at jdk.proxy2/jdk.proxy2.$Proxy239.merge(Unknown Source)
	at org.springframework.data.jpa.repository.support.SimpleJpaRepository.save(SimpleJpaRepository.java:620)
	at jdk.internal.reflect.GeneratedMethodAccessor138.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.springframework.data.repository.core.support.RepositoryMethodInvoker$RepositoryFragmentMethodInvoker.lambda$new$0(RepositoryMethodInvoker.java:288)
	at org.springframework.data.repository.core.support.RepositoryMethodInvoker.doInvoke(RepositoryMethodInvoker.java:136)
	at org.springframework.data.repository.core.support.RepositoryMethodInvoker.invoke(RepositoryMethodInvoker.java:120)
	at org.springframework.data.repository.core.support.RepositoryComposition$RepositoryFragments.invoke(RepositoryComposition.java:516)
	at org.springframework.data.repository.core.support.RepositoryComposition.invoke(RepositoryComposition.java:285)
	at org.springframework.data.repository.core.support.RepositoryFactorySupport$ImplementationMethodExecutionInterceptor.invoke(RepositoryFactorySupport.java:628)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.doInvoke(QueryExecutorMethodInterceptor.java:168)
	at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.invoke(QueryExecutorMethodInterceptor.java:143)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:391)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:137)
	... 61 more
```

## 2023-08-04

PostgreSQL
```bash
2023-08-04 14:32:10.164 UTC [1340935]: [1-1] db=,user= LOG:  automatic vacuum of table "mirror_node.public.transaction": index scans: 0
pages: 0 removed, 6860771 remain, 0 skipped due to pins, 3402675 skipped frozen
tuples: 0 removed, 143801292 remain, 0 are dead but not yet removable, oldest xmin: 32075651
index scan not needed: 0 pages from table (0.00% of total) had 0 dead item identifiers removed
avg read rate: 20.455 MB/s, avg write rate: 12.618 MB/s
buffer usage: 3614077 hits, 3304740 misses, 2038629 dirtied
WAL usage: 5991807 records, 2038634 full page images, 17091114648 bytes
system usage: CPU: user: 42.51 s, system: 123.15 s, elapsed: 1262.20 s
```

Importer
```bash
2023-08-04T08:32:44.138-0600 INFO scheduling-5 c.h.m.i.p.b.BatchInserter Copied 20000 rows to transaction table in 32.95 s
2023-08-04T08:32:44.161-0600 INFO scheduling-5 c.h.m.i.p.b.BatchInserter Copied 34 rows to entity_temp table in 23.00 ms
2023-08-04T08:32:44.161-0600 INFO scheduling-5 c.h.m.i.p.r.e.s.SqlEntityListener Completed batch inserts in 33.65 s
2023-08-04T08:32:44.199-0600 INFO pool-8-thread-1 c.h.m.i.p.r.e.r.RedisEntityListener Finished notifying 1183 messages in 32.08 ms
2023-08-04T08:32:44.203-0600 INFO scheduling-5 c.h.m.i.p.b.BatchInserter Copied 4742 rows to crypto_transfer table in 35.85 ms
2023-08-04T08:32:44.220-0600 INFO scheduling-5 c.h.m.i.p.b.BatchInserter Copied 1183 rows to topic_message table in 16.92 ms
2023-08-04T08:32:46.031-0600 INFO scheduling-5 c.h.m.i.p.b.BatchInserter Copied 1185 rows to transaction table in 1.811 s
2023-08-04T08:32:46.040-0600 INFO scheduling-5 c.h.m.i.p.b.BatchInserter Copied 27 rows to entity_temp table in 9.351 ms
2023-08-04T08:32:46.041-0600 INFO scheduling-5 c.h.m.i.p.r.e.s.SqlEntityListener Completed batch inserts in 1.874 s
2023-08-04T08:32:46.092-0600 ERROR scheduling-5 c.h.m.i.p.r.RecordFileParser Error parsing file 2023-08-04T14_32_02.012205960Z.rcd.gz after 35.69 s org.springframework.orm.jpa.JpaSystemException: transaction timeout expired
 at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:320)
 at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:229)
 at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.translateExceptionIfPossible(AbstractEntityManagerFactoryBean.java:550)
 at org.springframework.dao.support.ChainedPersistenceExceptionTranslator.translateExceptionIfPossible(ChainedPersistenceExceptionTranslator.java:61)
 at org.springframework.dao.support.DataAccessUtils.translateIfNecessary(DataAccessUtils.java:242)
 at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:152)
 at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
 at org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.invoke(CrudMethodMetadataPostProcessor.java:164)
 at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
 at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
 at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
 at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:244)
 at jdk.proxy2/jdk.proxy2.$Proxy255.save(Unknown Source)
 at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.onEnd(SqlEntityListener.java:230)
 at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.onEnd(SqlEntityListener.java:90)
 at com.hedera.mirror.importer.parser.record.entity.CompositeRecordStreamFileListener.onEnd(CompositeRecordStreamFileListener.java:44)
 at com.hedera.mirror.importer.parser.record.entity.CompositeRecordStreamFileListener.onEnd(CompositeRecordStreamFileListener.java:29)
 at com.hedera.mirror.importer.parser.record.RecordFileParser.doParse(RecordFileParser.java:145)
 at com.hedera.mirror.importer.parser.record.RecordFileParser.doParse(RecordFileParser.java:47)
 at com.hedera.mirror.i…
```

### Hedera network

other

### Version

v0.86.0-beta1

### Operating system

None

--- a/hedera-mirror-graphql/src/test/resources/application.yml
+++ b/hedera-mirror-graphql/src/test/resources/application.yml
@@ -20,7 +20,6 @@ spring:
     placeholders:
       api-password: mirror_api_pass
       api-user: mirror_api
-      autovacuumFreezeMaxAgeInsertOnly: 100000
       db-name: ${hedera.mirror.graphql.db.name}
       db-user: ${hedera.mirror.graphql.db.username}
       idPartitionSize: 1000000000000000

--- a/hedera-mirror-grpc/src/test/resources/config/application.yml
+++ b/hedera-mirror-grpc/src/test/resources/config/application.yml
@@ -43,7 +43,6 @@ spring:
     placeholders:
       api-password: mirror_api_pass
       api-user: mirror_api
-      autovacuumFreezeMaxAgeInsertOnly: 100000
       db-name: ${hedera.mirror.grpc.db.name}
       db-user: ${hedera.mirror.grpc.db.username}
       idPartitionSize: 1000000000000000

--- a/hedera-mirror-importer/src/main/resources/application.yml
+++ b/hedera-mirror-importer/src/main/resources/application.yml
@@ -112,7 +112,6 @@ spring:
     placeholders:
       api-password: ${hedera.mirror.importer.db.restPassword}
       api-user: ${hedera.mirror.importer.db.restUsername}
-      autovacuumFreezeMaxAgeInsertOnly: 100000
       db-name: ${hedera.mirror.importer.db.name}
       db-user: ${hedera.mirror.importer.db.username}
       idPartitionSize: 1000000

--- a/hedera-mirror-importer/src/main/resources/db/migration/v1/R__autovacuum_insert_only_tables.sql
+++ b/hedera-mirror-importer/src/main/resources/db/migration/v1/R__autovacuum_insert_only_tables.sql
@@ -1,25 +1,113 @@
 -------------------
 -- autovacuum insert-only tables more frequently to ensure most pages are visible for index-only scans
--- prior to postgresql 13, this can be achieved by configuring a more aggressive anti-wraparound autovacuum
 -------------------
 
+-- reset table storage parameters set prior to postgresql 13
+alter table if exists crypto_transfer reset (
+    autovacuum_freeze_max_age,
+    autovacuum_freeze_table_age,
+    autovacuum_freeze_min_age,
+    log_autovacuum_min_duration
+    );
+
+alter table if exists token_transfer reset (
+    autovacuum_freeze_max_age,
+    autovacuum_freeze_table_age,
+    autovacuum_freeze_min_age,
+    log_autovacuum_min_duration
+    );
+
+alter table if exists transaction reset (
+    autovacuum_freeze_max_age,
+    autovacuum_freeze_table_age,
+    autovacuum_freeze_min_age,
+    log_autovacuum_min_duration
+    );
+
+-- auto vacuum per each account balance snapshot if there are at least 10000 accounts' balance inserted
+alter table if exists account_balance set (
+  autovacuum_vacuum_insert_scale_factor = 0,
+  autovacuum_vacuum_insert_threshold = 10000
+  );
+
+-- auto vacuum daily with 15-minute interval account balance files
+alter table if exists account_balance_file set (
+  autovacuum_vacuum_insert_scale_factor = 0,
+  autovacuum_vacuum_insert_threshold = 96
+  );
+
+-- based on 12 contract actions per smart contract transaction
+alter table if exists contract_action set (
+  autovacuum_vacuum_insert_scale_factor = 0,
+  autovacuum_vacuum_insert_threshold = 2160000
+  );
+
+-- base on average of 3 contract logs per smart contract transaction
+alter table if exists contract_log set (
+  autovacuum_vacuum_insert_scale_factor = 0,
+  autovacuum_vacuum_insert_threshold = 540000
+  );
+
+-- max 300 smart contract transactions per second
+alter table if exists contract_result set (
+  autovacuum_vacuum_insert_scale_factor = 0,
+  autovacuum_vacuum_insert_threshold = 180000
+  );
+
+-- based on average of 10 state changes per smart contract transaction
+alter table if exists contract_state_change set (
+  autovacuum_vacuum_insert_scale_factor = 0,
+  autovacuum_vacuum_insert_threshold = 1800000
+  );
+
+-- autovacuum every 10 minutes at 10K TPS with average of 4 crypto transfers per transaction
 alter table if exists crypto_transfer set (
-    autovacuum_freeze_max_age = ${autovacuumFreezeMaxAgeInsertOnly},
-    autovacuum_freeze_table_age = ${autovacuumFreezeMaxAgeInsertOnly},
-    autovacuum_freeze_min_age = 0,
-    log_autovacuum_min_duration = 0
-    );
+  autovacuum_vacuum_insert_scale_factor = 0,
+  autovacuum_vacuum_insert_threshold = 24000000
+  );
 
+-- adjust when there are higher number of ethereum transactions per second
+alter table if exists ethereum_transaction set (autovacuum_vacuum_insert_scale_factor = 0.1);
+
+-- autovacuum every 60 minutes with 2s interval record files
+alter table if exists record_file set (
+  autovacuum_vacuum_insert_scale_factor = 0,
+  autovacuum_vacuum_insert_threshold = 1800
+  );
+
+-- normally there is just one sidecar record per each record file
+alter table if exists sidecar_file set (
+  autovacuum_vacuum_insert_scale_factor = 0,
+  autovacuum_vacuum_insert_threshold = 1800
+  );
+
+-- threshold set based on recent stats so the table has about 2 to 3 automatic vacuums per day, adjust if there's more
+-- daily pending reward claim
+alter table if exists staking_reward_transfer set (
+  autovacuum_vacuum_insert_scale_factor = 0,
+  autovacuum_vacuum_insert_threshold = 4000
+  );
+
+-- autovacuum per each token balance snapshot if there are at least 10000 account token balance inserted
+alter table if exists token_balance set (
+  autovacuum_vacuum_insert_scale_factor = 0,
+  autovacuum_vacuum_insert_threshold = 10000
+  );
+
+-- autovacuum every 10 minutes at 10K TPS crypto transaction transaction with two token transfer rows per transaction
 alter table if exists token_transfer set (
-    autovacuum_freeze_max_age = ${autovacuumFreezeMaxAgeInsertOnly},
-    autovacuum_freeze_table_age = ${autovacuumFreezeMaxAgeInsertOnly},
-    autovacuum_freeze_min_age = 0,
-    log_autovacuum_min_duration = 0
-    );
+  autovacuum_vacuum_insert_scale_factor = 0,
+  autovacuum_vacuum_insert_threshold = 12000000
+  );
 
+-- autovacuum every 10 minutes at 10K TPS consensus submit message transactions
+alter table if exists topic_message set (
+  autovacuum_vacuum_insert_scale_factor = 0,
+  autovacuum_vacuum_insert_threshold = 6000000
+  );
+
+-- autovacuum every 10 minutes at 10K TPS
 alter table if exists transaction set (
-    autovacuum_freeze_max_age = ${autovacuumFreezeMaxAgeInsertOnly},
-    autovacuum_freeze_table_age = ${autovacuumFreezeMaxAgeInsertOnly},
-    autovacuum_freeze_min_age = 0,
-    log_autovacuum_min_duration = 0
-    );
+  autovacuum_vacuum_insert_scale_factor = 0,
+  autovacuum_vacuum_insert_threshold = 6000000
+  );

--- a/hedera-mirror-importer/src/main/resources/db/migration/v1/R__autovacuum_insert_only_tables.sql
+++ b/hedera-mirror-importer/src/main/resources/db/migration/v1/R__autovacuum_insert_only_tables.sql
@@ -36,13 +36,13 @@ alter table if exists account_balance_file set (
   autovacuum_vacuum_insert_threshold = 96
   );
 
--- based on 12 contract actions per smart contract transaction
+-- based on 12 contract actions per smart contract transaction and max 300 TPS
 alter table if exists contract_action set (
   autovacuum_vacuum_insert_scale_factor = 0,
   autovacuum_vacuum_insert_threshold = 2160000
   );
 
--- base on average of 3 contract logs per smart contract transaction
+-- base on average of 3 contract logs per smart contract transaction and max 300 TPS
 alter table if exists contract_log set (
   autovacuum_vacuum_insert_scale_factor = 0,
   autovacuum_vacuum_insert_threshold = 540000
@@ -54,7 +54,7 @@ alter table if exists contract_result set (
   autovacuum_vacuum_insert_threshold = 180000
   );
 
--- based on average of 10 state changes per smart contract transaction
+-- based on average of 10 state changes per smart contract transaction and max 300 TPS
 alter table if exists contract_state_change set (
   autovacuum_vacuum_insert_scale_factor = 0,
   autovacuum_vacuum_insert_threshold = 1800000
@@ -66,7 +66,7 @@ alter table if exists crypto_transfer set (
   autovacuum_vacuum_insert_threshold = 24000000
   );
 
--- adjust when there are higher number of ethereum transactions per second
+-- adjust when ethereum transaction TPS becomes higher
 alter table if exists ethereum_transaction set (autovacuum_vacuum_insert_scale_factor = 0.1);
 
 -- autovacuum every 60 minutes with 2s interval record files
@@ -75,14 +75,15 @@ alter table if exists record_file set (
   autovacuum_vacuum_insert_threshold = 1800
   );
 
--- normally there is just one sidecar record per each record file
+-- autovacuum every 60 minutes with 2s interval record files. Note normally there is just one sidecar record
+-- per each record file
 alter table if exists sidecar_file set (
   autovacuum_vacuum_insert_scale_factor = 0,
   autovacuum_vacuum_insert_threshold = 1800
   );
 
 -- threshold set based on recent stats so the table has about 2 to 3 automatic vacuums per day, adjust if there's more
--- daily pending reward claim
+-- daily staking reward claims
 alter table if exists staking_reward_transfer set (
   autovacuum_vacuum_insert_scale_factor = 0,
   autovacuum_vacuum_insert_threshold = 4000
@@ -94,7 +95,7 @@ alter table if exists token_balance set (
   autovacuum_vacuum_insert_threshold = 10000
   );
 
--- autovacuum every 10 minutes at 10K TPS crypto transaction transaction with two token transfer rows per transaction
+-- autovacuum every 10 minutes at 10K TPS crypto transfer transactions with two token transfer rows per transaction
 alter table if exists token_transfer set (
   autovacuum_vacuum_insert_scale_factor = 0,
   autovacuum_vacuum_insert_threshold = 12000000

--- a/hedera-mirror-rest-java/src/test/resources/application.yml
+++ b/hedera-mirror-rest-java/src/test/resources/application.yml
@@ -19,7 +19,6 @@ spring:
     placeholders:
       api-password: mirror_api_pass
       api-user: mirror_api
-      autovacuumFreezeMaxAgeInsertOnly: 100000
       db-name: ${hedera.mirror.restJava.db.name}
       db-user: ${hedera.mirror.restJava.db.username}
       idPartitionSize: 1000000000000000

--- a/hedera-mirror-rest/__tests__/integrationDbOps.js
+++ b/hedera-mirror-rest/__tests__/integrationDbOps.js
@@ -111,7 +111,6 @@ const flywayMigrate = async () => {
       "password": "${dbConnectionParams.password}",
       "placeholders.api-password": "${defaultDbConfig.password}",
       "placeholders.api-user": "${apiUsername}",
-      "placeholders.autovacuumFreezeMaxAgeInsertOnly": 100000,
       "placeholders.db-name": "${dbName}",
       "placeholders.db-user": "${dbConnectionParams.user}",
       "placeholders.idPartitionSize": 1000000000000000,

--- a/hedera-mirror-rosetta/test/db/db.go
+++ b/hedera-mirror-rosetta/test/db/db.go
@@ -247,10 +247,9 @@ func runFlywayMigration(pool *dockertest.Pool, network *dockertest.Network, para
 	}
 
 	args := map[string]string{
-		"password":                  params.password,
-		"placeholders.api-password": "mirror_api",
-		"placeholders.api-user":     "mirror_api_password",
-		"placeholders.autovacuumFreezeMaxAgeInsertOnly": "100000",
+		"password":                                      params.password,
+		"placeholders.api-password":                     "mirror_api",
+		"placeholders.api-user":                         "mirror_api_password",
 		"placeholders.idPartitionSize":                  "1000000000000000",
 		"placeholders.maxEntityId":                      "5000000",
 		"placeholders.maxEntityIdRatio":                 "2.0",

--- a/hedera-mirror-web3/src/test/resources/application.yml
+++ b/hedera-mirror-web3/src/test/resources/application.yml
@@ -20,7 +20,6 @@ spring:
     placeholders:
       api-password: mirror_api_pass
       api-user: mirror_api
-      autovacuumFreezeMaxAgeInsertOnly: 100000
       db-name: ${hedera.mirror.web3.db.name}
       db-user: ${hedera.mirror.web3.db.username}
       idPartitionSize: 1000000000000000

